src/order/compactly_generated.lean

```diff
@@ -120,6 +120,19 @@ begin
     use t, exact ⟨htS, by rwa ←htsup⟩, },
 end
 
+lemma finset_sup_compact_of_compact (s : finset α) (h : ∀ x ∈ s, is_compact_element x) :
+  is_compact_element (s.sup id) :=
```

Comment:
It looks like later on, you want to do this with functions other than `id`. Perhaps you could prove this where `h : ∀ x ∈ s, is_compact_element (f x)` implies `is_compact_element `(s.sup f)`, so that you can apply it directly to the span-singleton function?